spyder/api/panel.py

```diff
@@ -125,6 +131,27 @@ def paintEvent(self, event):
                 self.palette().windowText().color()))
             painter = QPainter(self)
             painter.fillRect(event.rect(), self._background_brush)
+        else:
+            raise NotImplementedError(
+                f'paintEvent method must be defined in {self}')
+
+    def sizeHint(self):
+        """
+        Return the widget size hint, override the Qt method.
+
+        Notes:
+            - This size hint will define the QSize, where the width and height
+            is defined.
```

Comment:
This size hint defines the QSize of the panel, i.e. it is where its width and height are defined.